Testing/unittest/JsonLd/JsonLdParserTests.cs

```diff
@@ -44,5 +45,66 @@ public void TestIssue122()
                     x.Predicate.As<IUriNode>().Uri.ToString().Equals("http://schema.org/spouse") &&
                     x.Object.As<IUriNode>().Uri.ToString().Equals("http://dbpedia.org/resource/Cynthia_Lennon"));
         }
+
+        public static IEnumerable<object[]> DateTimeValues
+        {
+            get
+            {
+                return new[] {
+                    new object[] { "not_a_date", "http://www.w3.org/2001/XMLSchema#dateTime" },
+                    new object[] { "2000-01-01T00:00:00", "http://www.w3.org/2001/XMLSchema#dateTime" },
+                    new object[] { "2000-01-01T00:00:00-00:00", "http://www.w3.org/2001/XMLSchema#dateTime" },
+                    new object[] { "2000-01-01T00:00:00+00:00", "http://www.w3.org/2001/XMLSchema#dateTime" },
+                    new object[] { "2000-01-01T00:00:00-01:00", "http://www.w3.org/2001/XMLSchema#dateTime" },
+                    new object[] { "2000-01-01T00:00:00+01:00", "http://www.w3.org/2001/XMLSchema#dateTime" },
+                    new object[] { "not_a_date", "http://example.com/datatype" },
+                    new object[] { "2000-01-01T00:00:00", "http://example.com/datatype" },
+                    new object[] { "2000-01-01T00:00:00-00:00", "http://example.com/datatype" },
+                    new object[] { "2000-01-01T00:00:00+00:00", "http://example.com/datatype" },
+                    new object[] { "2000-01-01T00:00:00-01:00", "http://example.com/datatype" },
+                    new object[] { "2000-01-01T00:00:00+01:00", "http://example.com/datatype" },
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(DateTimeValues), DisableDiscoveryEnumeration = false)]
+        public void DateTimeHandlingDiff(string dateTimeValue, string datatype)
```

Comment:
Please rename the test method to something more descriptive.

I know, it's not nearly perfect throughout dotNetRDF but we must start somewhere ;)